src/storage/mod.rs

```diff
@@ -281,6 +287,34 @@ where
             .map_err(|e| anyhow!(e))
     }
 
+    /// Read bitfield header.
+    pub async fn read_bitfield(&mut self) -> Result<Vec<u8>> {
+        let buf = self
+            .bitfield
+            .read(0, 32)
+            .await
+            .expect("read bitfield header");
+        let header = Header::from_vec(&buf).unwrap();
+
+        // khodzha:
+        // TODO: we should handle eof vs errors here somehow but idk how to do that
```

Comment:
You could match on `Err(EOF)` and do something else than `Err(_)` in general.

Reply:
correct me if im wrong but i dont think this is possible right now
we erase errors into `Box<dyn std::error::Error + Sync + Send>`
https://github.com/datrs/random-access-disk/blob/1eb7cec052d8782f001e2150b45f6bc544e38c77/src/lib.rs#L38-L39